gen/src/main/java/com/softlayer/api/gen/ClassWriter.java

```diff
@@ -377,8 +379,24 @@ public ClassWriter emitType() throws IOException {
         
         // Now the service
         if (!type.meta.noservice) {
-            if (type.meta.properties.containsKey("id")) {
-                if (type.meta.properties.containsKey("globalIdentifier")) {
+
+            // Check if the type or any of its' parent types have id or globalIdentifier properties
+            Boolean containsId = false;
+            Boolean containsGlobalIdentifier = false;
+            Meta.Type searchType = type.meta;
+            while (searchType != null) {
+                if (searchType.properties.containsKey("id")) {
+                    containsId = true;
+                    if (searchType.properties.containsKey("globalIdentifier")) {
+                        containsGlobalIdentifier = true;
+                    }
+                    break;
+                }
+                searchType = meta.types.get(searchType.base);
+            }
```

Comment:
Usually preferred to use the non-boxed primitive `boolean` instead of `Boolean`. To write it another way:

```java
boolean containsId = false;
boolean containsGlobalIdentifier = false;
Meta.Type searchType = type.meta;
while (searchType != null && !containsId) {
    containsId = searchType.properties.containsKey("id");
    containsGlobalIdentifier = searchType.properties.containsKey("globalIdentifier");
    searchType = meta.types.get(searchType.base)
}
```

Meh, it works regardless. Good stuff.